internal/role/p2p/k8s.go

```diff
@@ -37,21 +37,187 @@ type K8sNode interface {
 	Distro() string
 }
 
+// configState holds the parsed configuration state.
+type configState struct {
+	k3sEnabled        bool
+	k3sAgentEnabled   bool
+	k0sEnabled        bool
+	k0sWorkerEnabled  bool
+	k3sDisabled       bool
+	k3sAgentDisabled  bool
+	k0sDisabled       bool
+	k0sWorkerDisabled bool
+	k3sBinAvailable   bool
+	k0sBinAvailable   bool
+	p2pConfigured     bool
+	p2pAutoEnabled    bool
+	p2pRoleSpecified  bool
+}
+
+// parseConfigState extracts and validates configuration state.
+func parseConfigState(c *providerConfig.Config) (*configState, error) {
+	state := &configState{
+		// Check if any k8s distro is explicitly enabled
+		k3sEnabled:       c.K3s.Enabled != nil && *c.K3s.Enabled,
+		k3sAgentEnabled:  c.K3sAgent.Enabled != nil && *c.K3sAgent.Enabled,
+		k0sEnabled:       c.K0s.Enabled != nil && *c.K0s.Enabled,
+		k0sWorkerEnabled: c.K0sWorker.Enabled != nil && *c.K0sWorker.Enabled,
+
+		// Check if any k8s distro is explicitly disabled
+		k3sDisabled:       c.K3s.Enabled != nil && !*c.K3s.Enabled,
+		k3sAgentDisabled:  c.K3sAgent.Enabled != nil && !*c.K3sAgent.Enabled,
+		k0sDisabled:       c.K0s.Enabled != nil && !*c.K0s.Enabled,
+		k0sWorkerDisabled: c.K0sWorker.Enabled != nil && !*c.K0sWorker.Enabled,
+
+		// Check if binaries are available
+		k3sBinAvailable: utils.K3sBin() != "",
+		k0sBinAvailable: utils.K0sBin() != "",
+
+		// Check if p2p is configured
+		p2pConfigured:    c.P2P != nil,
+		p2pAutoEnabled:   c.P2P != nil && c.P2P.Auto.Enable != nil && *c.P2P.Auto.Enable,
+		p2pRoleSpecified: c.P2P != nil && c.P2P.Role != "",
+	}
+
+	// Validate conflicting configurations
+	if state.p2pAutoEnabled && state.p2pRoleSpecified {
+		return nil, errors.New("cannot specify both p2p.auto.enabled=true and p2p.role, they contradict each other")
+	}
+
+	return state, nil
+}
+
+// checkExplicitDisables validates if any k8s components are explicitly disabled.
+func checkExplicitDisables(state *configState) error {
```

Comment:
Why would an explicit disable [produce an error](https://github.com/kairos-io/provider-kairos/pull/800/files#diff-4532f6e4027af7920f7181ad18f73baad0a426dcb68fc7b4602ca2939fba3154R197-R199)? Isn't that a valid user action? And immediately after the check, we call `handleExplicitEnables`. If something was explicitly disabled, we wouldn't reach that `handleExplicitEnables`, no?